src/pydvl/utils/numeric.py

```diff
@@ -134,6 +156,71 @@ def random_powerset(
         total += 1
 
 
+def random_powerset_group_conditional(
+    s: NDArray[T],
+    groups: NDArray[np.int_],
+    min_elements_per_group: int = 1,
+    seed: Optional[Seed] = None,
+) -> Generator[NDArray[T], None, None]:
+    """Draws random group-conditional subsets from `s`.
+
+    It is ensured that in each sampled set, each unique group is represented at
+    least `min_elements` times. The groups are specified as integers for all
+    elements of the set separately.
+
+    Args:
+        s: Vector of size N representing the set to sample elements from.
+        groups: Vector of size N containing the group as an integer for each
+            element.
+        min_elements_per_group: The minimum number of elements for each group.
+        seed: Either an instance of a numpy random number generator or a seed
+            for it.
+
+    Returns:
+        Generated draw from the powerset of s with `min_elements` of each group.
+
+    Raises:
+        ValueError: If `s` and `groups` are of different length or
+            `min_elements` is smaller than 0.
+    """
+    if len(groups) != len(s):
+        raise ValueError("Set and labels have to be of same size.")
+
+    if min_elements_per_group < 0:
+        raise ValueError(
+            f"Parameter min_elements={min_elements_per_group} needs to be bigger or equal to 0."
+        )
+
+    if min_elements_per_group == 0:
+        logger.warning(
+            "It is recommended to ensure at least one element of each group is"
```

Comment:
Recommended for what? This comment is coupled to cs Shapley, but the function available in a common module